### ISSUE TITLE
Unsupported Operator Error aten::numpy_T

### ISSUE THREAD

**Issue by TheSeriousProgrammer** [microsoft/nni#5474]
**Describe the issue**:
I was trying to prune this custom model of mine
```
class ResNetArc_Classifier(nn.Module):
    def __init__(self, model_type:str="resnet50", class_count:int=1000):
        super(ResNetArc_Classifier, self).__init__()
        resnet_model_fn = getattr(torchvision.models, model_type)
        self.inp_batch_norm = nn.BatchNorm2d(1)
        self.feature_network = resnet_model_fn()
        self.feature_network.conv1 = nn.Conv2d(1, 64, kernel_size=7, stride=2, padding=3, bias=False)
        self.feature_network.fc = nn.Identity()
    
        self.bn_final = nn.BatchNorm1d(2048)

        self.mapper = nn.Linear(2048, class_count, bias=False)
        
        self.s = 8
        self.m = 0
        self.update_margin(self.m, self.s)


    def update_margin(self, m = 0 , s=8):
        self.s = s
        self.m = m
        self.cos_m = math.cos(m)
        self.sin_m = math.sin(m)

    def forward(self, x ,y=None):
        assert self.s!=0

        x_norm = self.inp_batch_norm(x)
        feat = self.feature_network(x_norm)
        feat_norm = self.bn_final(feat)
        feat_l2 = F.normalize(feat_norm, dim=1)
        
        cosine_dist = torch.matmul(
            feat_l2,
            F.normalize(self.mapper.weight.T, dim=0)
        )

        sine_dist = torch.sqrt(1 - torch.square(cosine_dist))#.clamp(-1,1)

        out_dist = cosine_dist

        if y!=None :
        
            margined_cosine_dist = cosine_dist*self.cos_m - sine_dist*self.sin_m # cos(a+b) eqn
        
            margined_cosine_dist = torch.where(
                cosine_dist>0, 
                margined_cosine_dist, 
                cosine_dist
            ) # easy margin

            out_dist = margined_cosine_dist*y + cosine_dist*(1-y)
        
        out = self.s * out_dist

        return 8 * out
````

**Environment**:
- NNI version: 2.10
- Training service (local|remote|pai|aml|etc): local
- Client OS: 
- Server OS (for remote mode only):
- Python version: python3.8
- PyTorch/TensorFlow version: '1.12.0+cu113
- Is conda/virtualenv/venv used?: yes
- Is running in Docker?: no

**Log message**:
```
[2023-03-23 18:37:35] simulated prune pytorch_model.feature_network.layer4.2.conv2 remain/total: 500/512
[2023-03-23 18:37:35] simulated prune pytorch_model.feature_network.layer4.2.conv3 remain/total: 1997/2048
[2023-03-23 18:37:38] start to speedup the model
[2023-03-23 18:37:39] infer module masks...
[2023-03-23 18:37:39] Update mask for pytorch_model.inp_batch_norm
[2023-03-23 18:37:39] Update mask for pytorch_model.aten::numpy_T.180
[2023-03-23 18:37:39] ERROR: aten::numpy_T is not Supported! Please report an issue at https://github.com/microsoft/nni. Thanks
~
[2023-03-23 18:37:39] Update mask for pytorch_model.aten::rsub.195
[2023-03-23 18:37:39] Update mask for pytorch_model.feature_network.conv1
[2023-03-23 18:37:40] Update mask for pytorch_model.aten::norm.181<!--
```

Note that I have omitted all linear layers from compression by the following config

```
    pruner = LotteryTicketPruner(
        pl_model,
        [
            {
                "op_types":["Conv2d"],
                "total_sparsity":0.4,
            },
            {
                "exclude":True,
                "op_types":["Linear"]
            }
        ],
        "l2",
        evaluator=evaluator,
        speedup=True,
        total_iteration=20,
        log_dir = "./nni_outputs"
    )
    pruner.compress()
```

Would the issue have risen from the matmul operation? How can I fix the same

Thanks for creating this amazing repo though !!

**Comment by J-shang** [comment]
hello @TheSeriousProgrammer , this may becuase `numpy` is used in `model.feature_network`, you could check it, replace numpy op with torch op can avoid this.

**Comment by TheSeriousProgrammer** [comment]
Thanks for the response, I didnt use any nnumpy operator to my knowledge, is there any way to trace the location of the operator?

**Comment by J-shang** [comment]
please have a try with speedup v2 to find if there also has this issue, `python -m pip install --extra-index-url https://test.pypi.org/simple/ --no-cache-dir nni==3.0a0`,
then `from nni.compression.pytorch.speedup.v2 import ModelSpeedup`.

could you find the `feature_network (resnet50?)` code you used?

**Comment by Lijiaoa** [comment]
Do you still have this issue? Or any updates for it? @TheSeriousProgrammer

**Comment by TheSeriousProgrammer** [comment]
Hi the feature network code is the direct resnet50 code used in torchvision pytorch "https://pytorch.org/vision/main/_modules/torchvision/models/resnet.html#resnet50"

As an update, I reconfigured my pruning code to target the feature_network alone, and it worked !! i.e the target model passed was only the feature network

Any ideas on why the error could have occured?

I am attaching the modified snippet here :

```
from simple_trainer import AudioClassifierVectorDatasetPL, LightningWordClassifier
from pytorch_lightning import seed_everything
import torch
import tqdm
from nni.compression.pytorch.utils.counter import count_flops_params

from nni.compression.pytorch.pruning import (
    LinearPruner,
    AGPPruner,
    LotteryTicketPruner
)

seed_everything(4)

pl_model = LightningWordClassifier()
pl_model = pl_model.load_from_checkpoint("/home/captain-america/external_disk/.eff/AttentiveMobileWord-Trainer/resnet_50_noise/epoch=49-val_top1=0.7217.ckpt")
pl_model = pl_model.to("cuda:0")

pl_loader = AudioClassifierVectorDatasetPL()
pl_loader.setup(23)

train_loader = pl_loader.train_dataloader()
test_loader = pl_loader.test_dataloader()
iterations = 0

original_flops, _, _  = count_flops_params(pl_model.pytorch_model.feature_network, torch.rand(2,1,149,64).to("cuda:0"), verbose=False) 

def evaluator(feature_network):
    global iterations
    pl_model.pytorch_model.feature_network = feature_network
    pl_model.eval()

    output = 0
    count = 0
    with torch.autocast(device_type="cuda", dtype=torch.float16, cache_enabled=False):
        for batch in tqdm.tqdm(test_loader) :
            batch = [batch[0].to("cuda:0"), batch[1].to("cuda:0")]
            x_norm = pl_model.min_max_normalize(batch[0])
            logits = pl_model.forward(x_norm)
            accuracies = pl_model.topk_accuracy(logits, torch.squeeze(batch[1]), 101, mode = "test")
            output += accuracies["test_top1"]
            count+=1
    print(output, count)
    fraction = output/count
    iterations+=1

    current_flops, _, _  = count_flops_params(pl_model.pytorch_model.feature_network, torch.rand(2,1,149,64).to("cuda:0"), verbose=False) 
    torch.save(pl_model, f"workspace/model_{iterations}_{100*current_flops/original_flops:.2f}%_{100*fraction:.4f}%.pt")
    return fraction

current_accuracy = evaluator(pl_model.pytorch_model.feature_network)
print("Current_accuracy:",100*current_accuracy)

def finetuner(feature_network):
    pl_model.pytorch_model.feature_network = feature_network
    pl_model.train()


    with torch.autocast(device_type="cuda", dtype=torch.float16, cache_enabled=False,):

        optimizer = torch.optim.Adam(
            pl_model.parameters(), 
            lr=1e-3,
            eps=1e-4
        )

        for batch in tqdm.tqdm(train_loader):
            optimizer.zero_grad()
            batch = [batch[0].to("cuda:0"), batch[1].to("cuda:0")]
            loss = pl_model.training_step(batch, 0)
            loss.backward()
            optimizer.step()
    evaluator(pl_model.pytorch_model.feature_network)

#finetuner(pl_model.pytorch_model.feature_network)

kw_args = {
    "pruning_algorithm":"l2",
    "total_iteration":50,
    "evaluator":evaluator,
    "finetuner":finetuner,
    "speedup":True,
    "dummy_input":torch.rand((2,1,149,64)).to("cuda:0")
}

pruner = LotteryTicketPruner(
    pl_model.pytorch_model.feature_network,
    [
        {
            "op_types":["Conv2d"],
            "total_sparsity":0.5
        },
        {
            "exclude":True,
            "op_names":"layer4.2.conv3"
        }
    ],
    **kw_args
)

pruner.compress()
```

**Comment by J-shang** [comment]
hello @TheSeriousProgrammer , please install NNI from master branch and have a try the following code with model speedup v2, I modify the `F.normalize` to a customized `normalize` to avoid a strange error (will fix it in the future):

```
import math
import torch
import torch.nn as nn
import torchvision


def normalize(input, dim):
    return input / torch.norm(input, 2.0, dim=dim, keepdim=True).clamp_min(1e-12).expand_as(input)


class ResNetArc_Classifier(nn.Module):
    def __init__(self, model_type:str="resnet50", class_count:int=1000):
        super(ResNetArc_Classifier, self).__init__()
        resnet_model_fn = getattr(torchvision.models, model_type)
        self.inp_batch_norm = nn.BatchNorm2d(1)
        self.feature_network = resnet_model_fn()
        self.feature_network.conv1 = nn.Conv2d(1, 64, kernel_size=7, stride=2, padding=3, bias=False)
        self.feature_network.fc = nn.Identity()
    
        self.bn_final = nn.BatchNorm1d(2048)

        self.mapper = nn.Linear(2048, class_count, bias=False)
        
        self.s = 8
        self.m = 0
        self.update_margin(self.m, self.s)


    def update_margin(self, m = 0 , s=8):
        self.s = s
        self.m = m
        self.cos_m = math.cos(m)
        self.sin_m = math.sin(m)

    def forward(self, x ,y=None):
        assert self.s!=0

        x_norm = self.inp_batch_norm(x)
        feat = self.feature_network(x_norm)
        feat_norm = self.bn_final(feat)
        feat_l2 = feat_norm / torch.norm(feat_norm, 2.0, dim=1, keepdim=True).clamp_min(1e-12).expand_as(feat_norm)
        feat_l2 = normalize(feat_norm, dim=1)
        
        cosine_dist = torch.matmul(
            feat_l2,
            normalize(self.mapper.weight.T, dim=0)
        )

        sine_dist = torch.sqrt(1 - torch.square(cosine_dist))#.clamp(-1,1)

        out_dist = cosine_dist

        if y!=None :
        
            margined_cosine_dist = cosine_dist*self.cos_m - sine_dist*self.sin_m # cos(a+b) eqn
        
            margined_cosine_dist = torch.where(
                cosine_dist>0, 
                margined_cosine_dist, 
                cosine_dist
            ) # easy margin

            out_dist = margined_cosine_dist*y + cosine_dist*(1-y)
        
        out = self.s * out_dist

        return 8 * out


from nni.compression.pytorch.pruning import L1NormPruner
from nni.compression.pytorch.speedup.v2 import ModelSpeedup

config_list = [
    {
        "op_types":["Conv2d"],
        "total_sparsity":0.5
    },
    {
        "exclude": True,
        "op_names": ["feature_network.layer4.2.conv3"]
    }
]

model = ResNetArc_Classifier()
pruner = L1NormPruner(model, config_list)
_, masks = pruner.compress()
pruner._unwrap_model()

ModelSpeedup(model, torch.rand((2,1,149,64)), masks).speedup_model()
```

If it works, you could modify the following code
https://github.com/microsoft/nni/blob/a68cb047d4bd3b09ca0ddcf2f8fc4149204a9070/nni/compression/pytorch/pruning/basic_scheduler.py#L15
to
`from nni.compression.pytorch.speedup.v2 import ModelSpeedup`
to use v2 version in LotteryTicketPruner

**Comment by Lijiaoa** [comment]
any updates? @TheSeriousProgrammer
